src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -767,177 +633,105 @@ namespace winrt::Microsoft::Terminal::Control::implementation
+        // IMPORTANT! Set this callback up sooner than later. If we do it
+        // after Enable, then it'll be possible to paint the frame once
+        // _before_ the warning handler is set up, and then warnings from
+        // the first paint will be ignored!
+        _core->RendererWarning({ get_weak(), &TermControl::_RendererWarning });
 
-            // Tell the DX Engine to notify us when the swap chain changes.
-            // We do this after we initially set the swapchain so as to avoid unnecessary callbacks (and locking problems)
-            _renderEngine->SetCallback(std::bind(&TermControl::RenderEngineSwapChainChanged, this));
+        const auto coreInitialized = _core->InitializeTerminal(panelWidth,
+                                                               panelHeight,
+                                                               panelScaleX,
+                                                               panelScaleY);
+        if (!coreInitialized)
+        {
+            return false;
+        }
+        _interactivity->Initialize();
 
-            auto bottom = _terminal->GetViewport().BottomExclusive();
-            auto bufferHeight = bottom;
+        _AttachDxgiSwapChainToXaml(_core->GetSwapChain());
 
-            ScrollBar().Maximum(bufferHeight - bufferHeight);
-            ScrollBar().Minimum(0);
-            ScrollBar().Value(0);
-            ScrollBar().ViewportSize(bufferHeight);
-            ScrollBar().LargeChange(std::max<SHORT>(bufferHeight - 1, 0)); // scroll one "screenful" at a time when the scroll bar is clicked
+        // Tell the DX Engine to notify us when the swap chain changes. We do
+        // this after we initially set the swapchain so as to avoid unnecessary
+        // callbacks (and locking problems)
+        _core->SwapChainChanged({ get_weak(), &TermControl::RenderEngineSwapChainChanged });
 
-            localPointerToThread->EnablePainting();
+        auto bufferHeight = _core->BufferHeight();
 
-            // Set up blinking cursor
-            int blinkTime = GetCaretBlinkTime();
-            if (blinkTime != INFINITE)
-            {
-                // Create a timer
-                DispatcherTimer cursorTimer;
-                cursorTimer.Interval(std::chrono::milliseconds(blinkTime));
-                cursorTimer.Tick({ get_weak(), &TermControl::_CursorTimerTick });
-                cursorTimer.Start();
-                _cursorTimer.emplace(std::move(cursorTimer));
-            }
-            else
-            {
-                // The user has disabled cursor blinking
-                _cursorTimer = std::nullopt;
-            }
+        ScrollBar().Maximum(bufferHeight - bufferHeight);
+        ScrollBar().Minimum(0);
+        ScrollBar().Value(0);
+        ScrollBar().ViewportSize(bufferHeight);
+        ScrollBar().LargeChange(std::max(bufferHeight - 1, 0)); // scroll one "screenful" at a time when the scroll bar is clicked

```

Comment:
```suggestion
        ScrollBar().LargeChange(std::max(bufferHeight - 1, 0)); // scroll one "screen full" at a time when the scroll bar is clicked

```

Reply:
NAK on "screen full". Nobody says, like, "I ate two hand fulls of chocolate"

Reply:
full screen worth?